### DISCUSSION TITLE
motion_commander_example curiosities

### DISCUSSION THREAD

**Discussion by JamesDvp30** [bitcraze/discussions#979] [Q&A] [answered]
hi guys i bought recently the crazyflie drone, the flow deck and usb dongle. im planing to work with the api specially the motion_commander, as i saw in the lecture.  im workingin my final project of university with my partner using your drone. is about driving a drone using a brain computer interface(emotiv insight), Well im working to fly the drone with my mental waves and then im here to do some few question about the use of motion commander. 

i saw that it takeoff automaticately when you run the script, do the routine,  and land when its out of scope, can i do it manually right? like calling the function def take_off() or  def land(), any sugestions or tips? 

Moreover, i flew the drone inside our laboratory and it lose the control and crashed :), but its ok... does the drone is sensitive to interference? because there are a lot of Acces points, routers, etc... i flew it outdoor and worked perfectly.
im interest to know you point of view of my idea: i have on mind to send to the virtual machine api some  commands in words like : "right", "left", "up" and "down", throught a socket.
 
Moreover, im planning to make the drone flight in the direction i "want" thinking with the emotiv headset. have you some advice that can be for my help.i know i have to create a logic to secure my drone during flight, so i will limitate the comands by counting them and if a reach a X number of the same command i will force it to go the other side o coming back to te original position(when it take off) and so im planing to make a socket that send the mental commands to the VM ubuntu you provide. 

im open to suggestions, hope it interest you

**Comment by knmcguire** [comment, marked as answer]
Hi!  that is a very cool project. 

Yes that is possible to separately call take off and land. I refer to this discussion though for an example: https://github.com/orgs/bitcraze/discussions/969

What kind of positioning are you using? Then we can figure out what is going wrong.

**Reply by JamesDvp30** [reply]
hi, i have the flow deck. right now i will be testing the drone in my basement, to see how it goes

**Reply by JamesDvp30** [reply]
hi!! i hope you doing good,  i dont know if you can help me... but is there a command in the library to put the drone like "doing nothig" like stay where it is? plus, i was running this code on the api but the drone get stuck and doesnt move. i can provide you my code here:

Moreover, is there  a discussion abou sending a list of commands one after one ? i think it can guiide me in case im wrong

import logging
import time
import socket
import cflib.crtp
from cflib.crazyflie import Crazyflie
from cflib.crazyflie.syncCrazyflie import SyncCrazyflie
from cflib.positioning.motion_commander import MotionCommander
from cflib.utils import uri_helper
URI = uri_helper.uri_from_env(default='radio://0/80/2M/E7E7E7E7E7')
# Only output errors from the logging framework
logging.basicConfig(level=logging.ERROR)

def manejar_comandos():
    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    server_socket.bind(('0.0.0.0', 5000))
    server_socket.listen(1)
    print("socket server is active... waiting for connection")

    mc = None

    command_counters = {
        "right": 0,
        "left": 0,
        "up": 0,
        "down": 0
    }

    while True:
        client_socket, client_address = server_socket.accept()
        print("Client connected from:", client_address)

        if not mc:
            with SyncCrazyflie(URI, cf=Crazyflie(rw_cache='./cache')) as scf:
                mc = MotionCommander(scf)
                # mc.take_off()
                print(" drone has take off!")

                command_bytes = client_socket.recv(1024)  # Recibe datos en bytes
                command = command_bytes.decode('utf-8')  # Decodifica los datos a UTF-8
                print(f"command recieved: {command}")

                if command in command_counters:
                    if command_counters[command] < 3:
                        if command == "right":
                            # mc.right(0.5)
                            print("DRONE IS GOING TO THE RIGHT")
                            time.sleep(5)

                        elif command == "left":
                            print("DRONE IS GOING TO THE LEFT")
                            # mc.left(0.5, velocity=0.4)
                            time.sleep(5)
                        elif command == "up":
                            print("DRONE IS GOING UP")
                            # mc.up(0.3)
                            time.sleep(5)
                        elif command == "down":
                            print("DRONE IS GOING DOWN")
                            # mc.down(0.3)
                            time.sleep(5)
                        command_counters[command] += 1
                    else:
                        print(f"Command '{command}' the command has been executed more of 3 times")
                else:
                    print("CNo-reconized Command")
                    
                    
                    
THAT code above was the "server"



import socket
import time


def envioporsocket():
    host = '10.0.0.232' # you put here the ip of the VM
    port = 5000
    comandos = ["right", "left", "neutral", "up", "right"]

    client_socket = socket.socket(socket.AF_INET,socket.SOCK_STREAM)
    client_socket.connect((host,port))
    while True:
        for comano in comandos:
            codif = comano.encode('utf-8')
            client_socket.send((codif))
            print(f"sended command is: {comano}")
            time.sleep(8)

def main():

    envioporsocket()

if __name__ =='__main__':
    main()
this above is the client

**Reply by knmcguire** [reply]
Hi! 

It is a bit confusing now.. as you selected an answer already for this discussion. could you start a new one with your new question? Also be a bit more clear with explaining your issue as it is uncertain what is going wrong now in particular for your case.
